// module-account/lib/module/index.js
"use strict";

import LoginScreen from "./screens/login/LoginScreen.js";
import ProfileScreen from "./screens/profile/ProfileScreen.js";
import ProtectAccountScreen from "./screens/profile/protect_account/ProtectAccountScreen.js";
import TermScreen from "./screens/profile/term_screen/TermScreen.js";
import HelpCenterScreen from "./screens/profile/help_center_screen/HelpCenterScreen.js";
import FrequentlyQuestionScreen from "./screens/profile/frequently_questions_screen/FrequentlyQuestionScreen.js";
import ProfileChangePasswordScreen from "./screens/profile_detail/ProfileChangePasswordScreen.js";
import ProfileDetailEditScreen from "./screens/profile_detail/ProfileDetailEditScreen.js";
import ProfileDetailScreen from "./screens/profile_detail/ProfileDetailScreen.js";
import ForgetPasswordScreen from "./screens/forget_password_screen/ForgetPasswordScreen.js";
import RegisterScreen from "./screens/register/RegisterScreen.js";
export { LoginScreen, ProfileScreen, ProtectAccountScreen, TermScreen, HelpCenterScreen, FrequentlyQuestionScreen, ProfileChangePasswordScreen, ProfileDetailEditScreen, ProfileDetailScreen, ForgetPasswordScreen, RegisterScreen };
