// module-account/src/index.js
import LoginScreen from './screens/login/LoginScreen';
import ProfileScreen from './screens/profile/ProfileScreen';
import ProtectAccountScreen from './screens/profile/protect_account/ProtectAccountScreen';
import TermScreen from './screens/profile/term_screen/TermScreen';
import HelpCenterScreen from './screens/profile/help_center_screen/HelpCenterScreen';
import FrequentlyQuestionScreen from './screens/profile/frequently_questions_screen/FrequentlyQuestionScreen';
import SupportCenterScreen from './screens/profile/support_center/SupportCenterScreen';
import ProfileChangePasswordScreen from './screens/profile_detail/ProfileChangePasswordScreen';
import ProfileDetailEditScreen from './screens/profile_detail/ProfileDetailEditScreen';
import ProfileDetailScreen from './screens/profile_detail/ProfileDetailScreen';
import ForgetPasswordScreen from './screens/forget_password_screen/ForgetPasswordScreen';
import RegisterScreen from './screens/register/RegisterScreen';
import FrequentlyQuestionCloneScreen from './screens/profile/frequently_questions_clone/FrequentlyQuestionsCloneScreen'
export {
  LoginScreen,
  ProfileScreen,
  ProtectAccountScreen,
  TermScreen,
  HelpCenterScreen,
  FrequentlyQuestionScreen,
  SupportCenterScreen,
  ProfileChangePasswordScreen,
  ProfileDetailEditScreen,
  ProfileDetailScreen,
  ForgetPasswordScreen,
  RegisterScreen,
  FrequentlyQuestionCloneScreen
};
