// module-account/src/screens/profile/support_center/SupportCenter.styles.js

import { StyleSheet } from 'react-native';
import { Color } from '@theme/colors';
import { horizontal, vertical, textSize} from '@utils/scales';
import { screenWidth } from '@utils/platforms';

const RATE_IMAGE = 168.55 / (327 - 20 * 2);
const WIDTH_IMAGE = screenWidth;
const HEIGHT_IMAGE = WIDTH_IMAGE * RATE_IMAGE;
export const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: Color.white
    },
    titleScreen: {
    },
    line: {
        height: 15,
        backgroundColor: Color.color_border,
        // trên dưới 
      //  marginVertical: 20,
    },
    header: {
        padding: 24,
        flex:1,
        //marginVertical: vertical(10),
       
    },
    imgContainer: {
        padding: 24,
        width: WIDTH_IMAGE,
        height: HEIGHT_IMAGE,
        alignSelf: 'center',
      },
      img: {
        width: WIDTH_IMAGE,
        height: HEIGHT_IMAGE,
        alignSelf: 'center',
      },
    footer: {
        marginVertical: vertical(10),
        border: 10,
    },
    itemMenu: {
        flexDirection: 'row',                    
        width: screenWidth - horizontal(24) * 2,
        alignSelf: 'center',                    
        alignItems: 'center',                    
        paddingVertical: 8,                      
      },
    textMenu: {
        marginHorizontal: 5, // căn trái phải
        marginVertical: 10, // căn trên dưới
        marginLeft : 10, // căn chữ với icon
        fontWeight: '700',
        fontSize: textSize(14),
        lineHeight: 36,
      },
      iconRight: {
        marginLeft: 'auto',
      },
      title: {
        lineHeight: textSize(36),
        fontSize: textSize(24),
        fontWeight: '700',
      },
      preview: {
        lineHeight: textSize(24),
        fontSize: textSize(14),
        fontWeight: '400',
        color: Color.text_color_hover,
        marginVertical: 10,
      },
      textInput: {
        borderRadius: 16,
        marginTop: 20,
      },
});
